src/ucp/core/ucp_worker.c

```diff
@@ -1346,7 +1356,7 @@ ucs_status_t ucp_worker_wait(ucp_worker_h worker)
     }
 
     if (worker->flags & UCP_WORKER_FLAG_EXTERNAL_EVENT_FD) {
-        pfd = ucs_alloca(sizeof(*pfd) * worker->context->num_tls);
+        pfd = ucs_alloca(sizeof(*pfd) * ucs_list_length(&worker->ifaces));
```

Comment:
can't you use `context->num_tls` here? Would be good to avoid calling `ucs_list_length`

Reply:
I can for now, because now the list and the array hold the same wifaces. But after I add the sockaddr wifaces the list will have wifaces that the array doesn't.

Reply:
so maybe it is worth to maintain an iface counter separately? Otherwise you would need to go thru the whole list for every wait call.

Reply:
Why go over the whole list?

Reply:
ucs_list_length does that